scripts/delete_from_repository.py

```diff
@@ -0,0 +1,144 @@
+    res = requests.get(task_url, auth=auth).json()
+    while res['state'] not in ('failed', 'completed'):
+        time.sleep(3)
+        res = requests.get(task_url, auth=auth).json()
+    return res
+
+
+def publish_repo(base_url: str, repo_href: str, auth: HTTPBasicAuth):
+    full_url = urljoin(base_url, '/pulp/api/v3/publications/rpm/rpm/')
+    task_href = requests.post(
+        full_url, json={'repository': repo_href}, auth=auth).json()['task']
+    return wait_for_task(base_url, task_href, auth)
+
+
+def setup_logging(verbose: bool = False):
+    log_level = logging.DEBUG if verbose else logging.INFO
+    logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s',
+                        level=log_level, datefmt='%Y-%m-%d %H:%M:%S')
+
+
+def main():
+    packages_endpoint = '/pulp/api/v3/content/rpm/packages/'
+    arguments = parse_args(sys.argv[1:])
+    setup_logging(verbose=arguments.verbose)
+    logger = logging.getLogger(PROG_NAME)
+    pulp_auth = HTTPBasicAuth(
+        settings.pulp_user, settings.pulp_password)
+    pulp_host = settings.pulp_host
+
+    repo = get_repository(
+        arguments.repository_name, arguments.architecture)
+    logger.debug('Repository: %s', repo)
+    repo_info = requests.get(
+        urljoin(pulp_host, repo.pulp_href), auth=pulp_auth).json()
+    # Looking for packages that belong to the latest repository version
+    query_params = {
+        'repository_version': repo_info['latest_version_href'],
+        'fields': ['location_href', 'pulp_href'],
+    }
+    # Check how many packages is in the repo to get them the in 1 request
+    logger.info('Getting repository packages')
+    pulp_response = requests.get(
+        urljoin(pulp_host, packages_endpoint), params=query_params,
+        auth=pulp_auth).json()
+    package_count = pulp_response['count']
+    repo_packages = pulp_response['results']
+    logger.debug('Packages count: %d', package_count)
+    if package_count > 100:
+        query_params['limit'] = package_count + 1
+        repo_packages = requests.get(
+            urljoin(pulp_host, packages_endpoint),
+            params=query_params, auth=pulp_auth).json()['results']
+    logger.info('Got all packages')
+
+    logger.info('Compiling list of packages hrefs to delete')
+    to_remove = []
+    # location href may be weird, so we need double loop
+    # to properly detect packages being in the repo
+    for pkg in arguments.packages:
+        for pulp_pkg in repo_packages:
+            if pkg == pulp_pkg['location_href'] \
+                    or pkg in pulp_pkg['location_href']:
+                to_remove.append(pulp_pkg['pulp_href'])
+    logger.info('List is compiled')
+
+    if not to_remove:
+        logger.warning('Nothing to delete, packages %s'
+                       ' were not found in the repository',
+                       str(arguments.packages))
+        return
+
+    modify_url = urljoin(urljoin(pulp_host, repo.pulp_href), 'modify/')
```

Comment:
we already make this repo url above `urljoin(pulp_host, repo.pulp_href)`, we can reuse it